docs/handbooks/language-implementation.md

```diff
@@ -0,0 +1,144 @@
+# Language Implementation Handbook
+
+## Foreword
+
+Implementing a new language in *jsii* is not just a matter of implementing code
+generation: mapping the *[jsii type system]* to a new programming language takes
+some amounts of API design, finding ways to represent APIs in the most idiomatic
+possible way. This is a craft that often requires trial and error, and the best
+(if not only) way to validate a proposal is to put it in front of users and
+seek feedback. As a consequence, this endeavor should be expected to span
+months, not weeks.
+
+
+## Scoping & Planning
+
+The first step of most successful projects is to start by scoping work out and
+establishing a baseline plan to execute on. For contributors not yet familiar
+with *jsii*, the [specification] document is a great place to start. In
+particular, the [New Language Intake] document provides a high-level view of the
+recommended process for implementing new language support.
+
```

Comment:
I would add a table here that can be used as a TOC and summarizes the recommended activities and some rough estimation of how long they take in developer days/weeks